docs/dev/reference/services.md

```diff
@@ -492,6 +492,52 @@ class Example
 }
 ```
 
+## InsertTagParser
+
+{{< version "4.13" >}}
+
+This service let you replace inserttags in your string content.
+
+Some methods return a `ChunkedText` instance. The `ChunkedText` container was created to keep the surrounding text containing the insert tags separate from the replacements made by the insert tag parser. It is used for example in the twig escaper to skip encoding on inserttag replacement results.
+
+```php
+use Contao\CoreBundle\InsertTag\ChunkedText;
+use Contao\CoreBundle\InsertTag\InsertTagParser;
+
+class Example
+{
+    private InsertTagParser $insertTagParser;
+
+    public function __invoke(string $buffer): string
+    {
+        // Return a string and should be used in HTML context when the replaced result is sent as a response to the client. It will allow for ESI tags that can improve caching behaviour.
+        $resultReplace = $this->insertTagParser->replace($buffer);
+
+        // Return a string and should be used when the result is not sent to a client ( e.g. when using `$email->subject()`).
+        $resultReplaceInline = $this->insertTagParser->replaceInline($buffer);
+
+        // Return ChunkedText instance and should be used in HTML context when the replaced result is sent as a response to the client. It will allow for ESI tags that can improve caching behaviour.
```

Comment:
```suggestion
        // Return ChunkedText instance and should be used in HTML context when the replaced result is sent as a 
        // response to the client. It will allow for ESI tags that can improve caching behaviour.
```